streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -403,10 +403,10 @@ public synchronized void close() {
         }
 
         dbAccessor.close();
+        db.close();
```

Comment:
Can we add a comment and maybe link to RocksDB docs to clarify the correct order of close() calls? Otherwise, we might forget about it and reorder accidentally...

Also: this may affect more versions compared to the memory leak? Should we extract it into it's own PR to easily backport?

Reply:
Looking again at the docs https://github.com/facebook/rocksdb/wiki/RocksJava-Basics I am still not sure if the order is now completely correct.
The doc says ColumnFamilyHandle > RocksDB > DBOptions > ColumnFamilyOptions
but the code does ColumnFamilyHandle > ColumnFamilyOptions > DBOptions > RocksDB due to userSpecifiedOptions.
Additionally, only some are set to null afterwards, does this have a special reason?

Reply:
This PR is moving the `db.close()` to before `userSpecifiedOptions.close()` so aren't we now closing the db in the right order? That said, the docs do suggest we should flip the order in which `DBOptions` & `ColumnFamilyOptions` are closed in `RocksDBGenericOptionsToDbOptionsColumnFamilyOptionsAdapter#close` (which, side note, is an obscenely long name...)

But Matthias is right, this should be split into a separate PR (and well documented) -- will open another one

Also, I think on this branch everything *is* set to null -- but if you were looking at the 2.2 branch, yeah I did forget to set filter to null. We were rushing to get that in before 2.2.1 was cut 😄

Reply:
Thanks for splitting out https://github.com/apache/kafka/pull/7076 @ableegoldman 

Please review the new PR, too, @pkleindl

Reply:
Will do.
My question regarding the null setting was about the DBOptions & ColumnFamilyOptions in RocksDBGenericOptionsToDbOptionsColumnFamilyOptionsAdapter, why they are not set to null too.
As for the long name, it's probably a german language thing ;-)